dttools/src/jx.c

```diff
@@ -0,0 +1,320 @@
+/*
+Copyright (C) 2015- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "jx.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+struct jx_pair * jx_pair( struct jx *key, struct jx *value, struct jx_pair *next )
```

Comment:
For strict JSON, you are correct.  However, I am thinking of using this library to represent a superset of JSON, so I made things a bit more general.  For example, a non-string key would be handy for an external representation of an itable.  And, the symbol value type will allow for interesting substitutions.  It might be wise to add some options which control whether strict JSON or the superset is accepted...